### ISSUE TITLE
Feat: Add query result value to logs

### ISSUE THREAD

**Issue by MarkSRobinson** [argoproj/argo-rollouts#1719]
# Summary

I'd like to see the value of the AnalysisRun result put into the logs.  This would be similar to the AnalysisRun object in k8s, but would make it easier to analyze bugs via digging through the logs rather than relying on k8s recycling objects.  Specifically, I want the value from the query that Argo received.

# Use Cases

Tracking down bugs where Rollouts fail or succeed for unusual reasons.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by huikang** [comment]
Hi, @MarkSRobinson , could you clarify the problem a bit? 
> I'd like to see the value of the AnalysisRun result put into the logs.

Do you mean the argo-rollouts controller log?


>  This would be similar to the AnalysisRun object in k8s, but would make it easier to analyze bugs via digging through the logs rather than relying on k8s recycling objects.

could you provide any example to help understand the proposal? Thanks.

**Comment by MarkSRobinson** [comment]
Yes, I'd like to see the result in the controller log.

Ideally the log would look like
`<date> - AnalysisRun Complete - Status: Failed - Value: 25 - Object name: MyAnalysisRun - Namespace: MyNamespace`

Then I could dig into my logs to determine why a deployment failed.  This is generally easier for people than accessing kubernetes objects.

**Comment by huikang** [comment]
@MarkSRobinson , thanks for your explanation. The feature also makes it easier or any other CD tool like spinnaker to examine the analysis run results. Please take a look #1724 to see if it satisfy the requirement. 

```
   # sample log in the controller
   INFO[2021-12-20T00:06:45-05:00] Event(v1.ObjectReference{Kind:"AnalysisRun", Namespace:"default", Name:"prom-job-jcflh", UID:"96932355-9477-47d1-8a2d-7732b09ce355", APIVersion:"argoproj.io/v1alpha1", ResourceVersion:"61462", FieldPath:""}): type: 'Normal' reason: 'MeasurementSuccessful' Measurement Completed. Result: Successful. Value:  1

```

**Comment by agrawroh** [comment]
I agree that it make sense for us to provide a way so that we can have these results in some sort of persistent storage, making it easier to do post-analysis but at the same time I don't think controller logs are the best place to do this.

Log messages can change over time and it requires a lot of manual effort to keep everything in sync. I think an interface where you can define your own schema and have these results shipped to something like S3 would be more useful and desirable.

**Comment by MarkSRobinson** [comment]
I can see the value is exporting to some other tool for analysis.  My key use case is debugging when something goes wrong.  In my case, standard logs would be completely fine as an engineer could look through them on their own.

**Comment by huikang** [comment]
@agrawroh , I think the event log for the measurement is in relatively stable format like the analysis run event 

https://github.com/argoproj/argo-rollouts/blob/a3477cfbb85b40b0d5722a1f661d4dd261550fa4/analysis/analysis.go#L143-L150

**Comment by KipsasJaujoj** [comment]
This would be really useful for us as well

**Comment by reilly3000** [comment]
This would be really useful for debugging faulty measurements or conditions. I think it would make sense behind a debug log level.
